src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,27 @@
+package duke.command;
+
+import duke.tasklist.TaskList;
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.exceptions.TaskException;
+
+/**
+ * Gives command to execute an action
+ */
+public abstract class Command {
+
+    /**
+     * Exits duke if it detects bye command
+     */
+    public abstract boolean isExit();
+
```

Comment:
Perhaps you could have isExit() normally return false, but only override it in ByeCommand to return true?